keps/sig-testing/1746-stage-e2e-test-framework/README.md

```diff
@@ -0,0 +1,643 @@
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the
+functionality is accessed.
+
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real world usage
+- N installs
+- More rigorous forms of testing e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least 2 releases between beta and
+GA/stable, since there's no opportunity for user feedback, or even bug reports,
+in back-to-back releases.
+
+#### Removing a deprecated flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality which deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include [conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+This enhancement is not a conventional KEP as it does not introduce any user
+facing features, however, we plan to use the  conventional feature gate
+nomenclature to organize our work into stages.
+
+**Alpha**
+
+During the alpha stage, we will focus on decoupling the e2e test framework
+from core Kubernetes.
+We will consider this stage as completed the moment we are able to use
+import-boss to blanket disallow any core Kubernetes code from the e2e test framework.
+
+This is the work we plan to do during the 1.19 release cycle.
+
+**Beta**
+
+For the beta stage we will resume our work refactoring the e2e test framework,
+as stated in https://github.com/kubernetes/kubernetes/issues/76206.
+This stage will be completed when the e2e test framework only contains code for
+setting up and configuring e2e tests but no e2e test utility functions.
+All utility functions used for e2e testing have to be contained in appropriate
+e2e test framework subpackage.
+
+We will also work on surfacing the coming move of the e2e test framework onto
+staging to advise people that the e2e test framework is about to be pulled out
+of core Kubernetes and published as its own package.
+We will maintain core Kubernetes tests packages to reflect the work being done
+here.
```

Comment:
Will we start using staging e2e framework on k/k repo on this Beta phase?
I'd like to find when we will switch using staging one instead of the existing one and remove the existing framework code in this plan.